src/protocol/protocol.c

```diff
@@ -129,18 +129,17 @@ void PROTOCOL_Load(int no_dlg)
 {
     (void)no_dlg;
 #ifdef ENABLE_MODULAR
-    if(! PROTOCOL_HasModule(Model.protocol)) {
-        *loaded_protocol = 0;
+    *loaded_protocol = 0;
+    if(! PROTOCOL_HasModule(Model.protocol))
         return;
```

Comment:
The reason the code didn't do this originally is that we don't want to invalidate the protocol unless we are actually changing it.